### PR TITLE
STO doc :: Download Images from a Private Registry topic needs clarification re scanner templates vs. Security step DOC-3759

### PR DESCRIPTION
**Pull request by douglas-j-bothwell**
# Harness Developer Pull Request

- Preview link: [Configure STO to Download Images from a Private Registry](https://650daa142330da0ca59a368d--harness-developer.netlify.app/docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/download-images-from-private-registry)

- [DOC-3754 STO doc :: Download Images from a Private Registry topic needs clarification re scanner templates vs. Security step](https://harness.atlassian.net/browse/DOC-3754)

### CODE REVIEW

File: docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/download-images-from-private-registry.md [hunk 1]
```diff
@@ -97,16 +133,27 @@ USER 1000
 
    * `runner_registry_token`  — As needed
 
-   If you need to use a proxy server, you must also specify the following: 
+  :::note
+
+  `runner_registry` settings are not supported for scanner templates.
```
Comment (williamwissemann) [inline docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/download-images-from-private-registry.md:138]: I would maybe change this to read something like:

` These settings are ONLY supported by the Security step and are NOT supported by scanner template.`

 Just referring to `runner_registry` confused me (and I think I'm the one who wrote that)

Comment (douglas-j-bothwell) [reply docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/download-images-from-private-registry.md]: Updated to: These settings are supported by the **Security** step only. They are not supported by scanner templates.

File: docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/download-images-from-private-registry.md [hunk 2]
```diff
@@ -83,7 +94,32 @@ USER 1000
...
+
+Do the following if you're using a scanner template rather than a generic **Security** step. A scanner template is a **Security Test** step with a scanner-specific UI, such as [**Black Duck Hub**](/docs/security-testing-orchestration/sto-techref-category/black-duck-hub-scanner-reference), [**CodeQL**](/docs/security-testing-orchestration/sto-techref-category/codeql-scanner-reference#codeql-step-settings-for-sto-scans), and [**Snyk**](/docs/security-testing-orchestration/sto-techref-category/snyk-scanner-reference#snyk-step-configuration).
+
+1. In the stage where you're setting up the scanner template, go to the **Infrastructure** tab. 
+
+2. Under **Advanced**, go to **Override Image Connector (optional)** and select the connector to your private registry. You might need to scroll down to see this option.
+
+   ![](./static/override-image-connector.png)
+
+3. If you specified a `USER` in the Dockerfile for your scan image, configure the scan step to run as the the user:
```
Comment (SudheendraKatte) [inline docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/download-images-from-private-registry.md:112]:
```suggestion
3. If you specified a `USER` in the Dockerfile for your scan image, configure the scan step to run as the user:
```

File: docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/download-images-from-private-registry.md [hunk 3]
```diff
@@ -83,7 +94,32 @@ USER 1000
...
+
+   1. Open the scanner step and expand **Additional Configuration**. 
+   
+   2. Set the **Run as User** (`runAsUser`) setting to the user you specified in your Dockerfile.
+
+### Security step setup 
+
+Do the following if you're using a generic **Security** step for you scan.
```
Comment (SudheendraKatte) [inline docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/download-images-from-private-registry.md:120]:
```suggestion
Do the following if you're using a generic **Security** step for you scan:
```

File: docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/download-images-from-private-registry.md [hunk 4]
```diff
@@ -97,16 +133,27 @@ USER 1000
...
+
+2. If you need to use a proxy server, you must also specify the following: 
 
    * `http_proxy`  —  The hostname and port to use for proxied HTTP requests
   
    * `https_proxy`  —  The hostname and port to use for proxied HTTPS requests
 
    * `no_proxy`  — A comma-separated list of hosts to bypass the proxy
 
+3. If you specified a `USER` in the Dockerfile for your scan image, configure the scan step to run as the the user:
```
Comment (SudheendraKatte) [inline docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/download-images-from-private-registry.md:150]:
```suggestion
3. If you specified a `USER` in the Dockerfile for your scan image, configure the scan step to run as the user:
```

**Review by SudheendraKatte — APPROVED** [review]

**Pull request merged**